docs/branching-and-tagging.md

```diff
@@ -0,0 +1,82 @@
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+# Branching and Tagging
+
+## Overview
+
+Iceberg table metadata maintains a log of snapshots which represent the changes applied to a table.
+Snapshots are fundamental in Iceberg as they are the basis for reader isolation and time travel queries.
+For controlling metadata size and storage costs, Iceberg provides snapshot lifecycle management procedures such as [`expire_snapshots`](../../spark/spark-procedures/#expire-snapshots) for removing unused snapshots and no longer neccessary data files based on table snapshot retention properties.
+
+**For more sophisticated snapshot lifecycle management, Iceberg also supports branches and tags which are named references to snapshots.** Branches are independent lineages of snapshots and point to the head of the lineage.
+Branches and tags have independent lifecycles with regards to snapshot retention during `expireSnapshots`.
+
+Branching and tagging can be used for handling GDPR requirements and retaining important historical snapshots for auditing. 
+Branches can also be used as part of data engineering workflows, for enabling experimental branches for testing and validating new jobs.
+See below for some examples of how branching and tagging can facilitate these use cases.
+
+![Historical Tags](../img/historical-snapshot-tag.png)
+
+The above diagram shows an example of using tag to retain historical snapshots. 
+In this case a end of year tag is created for the final snapshot in the year, and the tagged snapshot has a retention of 5 years for compliance reasons. 
+
+![Audit Branch](../img/audit-branch.png)
+
+The above diagram shows an example of using an audit branch for validating a write workflow. 
+Writes are performed on a separate `audit-branch` independent from the main table history. 
+A validation workflow can validate (e.g. data quality) the state of `audit-branch`. 
+After validation, the main branch can be `fastForward` to the head of `audit-branch` to update the main table state. 
+
+Branches and tags have a maximum reference age property which control when the reference to the snapshot itself should be expired. Branches have retention properties which define the minimum number of snapshots to retain on a branch as well as the maximum age of individual snapshots to retain on the branch. 
+These properties are used when the expireSnapshots procedure is run. 
+For details on the algorithm for expireSnapshots, refer to the [spec](../../../spec#snapshot-retention-policy).
+
+## Usage 
```

Comment:
I think given the current changes, we can just make this a list of links instead of many different subsections.